### PR TITLE
Disable promote build in channel implementation

### PR DESCRIPTION
**Pull request by JohnTortugo**
Right now we don't need to do this build promotion here because Push to BAR step already does that due the default channel for the branch: https://github.com/dotnet/arcade-services/blob/master/src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs#L85

### CODE REVIEW

**Comment by riarenas** [comment]
Should we instead make promote-build tolerant of failures if the build is already in the channel? default channels can be disabled at any moment.

**Comment by JohnTortugo** [comment]
> Should we instead make promote-build tolerant of failures if the build is already in the channel? default channels can be disabled at any moment.

I'm OK with going with that approach as well. I'll need to add another API request, though.

FYI: default channels is integral for the current stage-based publishing approach. Stages are activated based on default channels information.

**Comment by JohnTortugo** [comment]
Address the feedback. Test build is here: https://dnceng.visualstudio.com/internal/_build/results?buildId=240540&view=results

**Comment by JohnTortugo** [comment]
Test build passed: https://dnceng.visualstudio.com/internal/_build/results?buildId=240540&view=logs&j=6dc017f3-b1b2-522f-af19-eea1dc9af249&t=1301f578-1e45-5c74-0401-a446527331cf&l=13

File: eng/common/templates/post-build/channels/internal-servicing.yml [hunk 1]
```diff
@@ -162,6 +162,6 @@ stages:
             darc gather-drop --non-shipping --continue-on-error --id $(BARBuildId) --output-dir $(Agent.BuildDirectory)/Temp/Drop/ --bar-uri https://maestro-prod.westus2.cloudapp.azure.com/ --password $(MaestroAccessToken) --latest-location
         enabled: false
 
-  - template: ../promote-build.yml
-    parameters:
+  - template: ../promote-build.yml
```
Comment (riarenas) [inline eng/common/templates/post-build/channels/internal-servicing.yml]: [nit] Whitespace

File: eng/common/post-build/promote-build.ps1 [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  [string] $MaestroEndpoint = "https://maestro-prod.westus2.cloudapp.azure.com",
+  [string] $ApiVersion = "2019-01-16"
+)
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version 2.0
+
+. $PSScriptRoot\..\tools.ps1
+
+function Get-Headers([string]$accept, [string]$barToken) {
```
Comment (riarenas) [inline eng/common/post-build/promote-build.ps1]: We have this Get-Headers function all over. Would you mind creating a cleanup issue for general refactoring of these scripts to see if it makes sense to reuse this?

File: eng/common/post-build/promote-build.ps1 [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  $headers = New-Object 'System.Collections.Generic.Dictionary[[String],[String]]'
+  $headers.Add('Accept',$accept)
+  $headers.Add('Authorization',"Bearer $barToken")
+  return $headers
+}
+
+try {
+  $maestroHeaders = Get-Headers 'application/json' $BarToken
+
+  # Get build info with info about which channels it has already been promoted to
```
Comment (riarenas) [inline eng/common/post-build/promote-build.ps1]:
```suggestion
  # Get info about which channels the build has already been promoted to
```

File: eng/common/post-build/promote-build.ps1 [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  if ($buildInfo.channels) {
+    foreach ($channel in $buildInfo.channels) {
+      if ($channel.Id -eq $ChannelId) {
+        Write-Host "The build with BAR ID $BuildId is already on channel $ChannelId!"
+        ExitWithExitCode 0
+      }
+    }
+  }
+
+  Write-Host "Build not promoted to channel $ChannelId. Promoting build ... "
```
Comment (riarenas) [inline eng/common/post-build/promote-build.ps1]:
```suggestion
  Write-Host "Build not present in channel $ChannelId. Promoting build ... "
```

Does that make sense?

File: eng/common/templates/post-build/channels/public-release.yml [hunk 5]
```diff
@@ -162,6 +162,6 @@ stages:
             darc gather-drop --non-shipping --continue-on-error --id $(BARBuildId) --output-dir $(Agent.BuildDirectory)/Temp/Drop/ --bar-uri https://maestro-prod.westus2.cloudapp.azure.com/ --password $(MaestroAccessToken) --latest-location
         enabled: false
 
-  - template: ../promote-build.yml
-    parameters:
+  - template: ../promote-build.yml
```
Comment (riarenas) [inline eng/common/templates/post-build/channels/public-release.yml]: [nit] same whitespace

**Pull request merged**